### PR TITLE
Task11 implement transaction

### PR DESCRIPTION
**Pull request by Platton96**

### CODE REVIEW

File: MoneySplitter.Services/API/QueryApiService.cs [hunk 1]
```diff
@@ -69,6 +69,8 @@ public class QueryApiService : IQueryApiService
             var contentResponce = await responseMessage.Content.ReadAsStringAsync();
             return JsonConvert.DeserializeObject<TContentResponce>(contentResponce);
         }
+
```
Comment (Vovanella95) [inline MoneySplitter.Services/API/QueryApiService.cs]: Empty lines

File: MoneySplitter.Services/API/TransactionsApiService.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        }
+
+        public async Task<ExecutionResult<IEnumerable<TransactionModel>>> GetAllUserTransactions()
+        {
+            var result = new ExecutionResult<IEnumerable<TransactionModel>>
+            {
+                IsSuccess = false
+            };
+
+            var apiUrlGetAllFriends = _apiUrlBuilder.GetAllUserTransactions();
```
Comment (Vovanella95) [inline MoneySplitter.Services/API/TransactionsApiService.cs]: allFriendsUrl

File: MoneySplitter.Services/API/TransactionsApiService.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+            IEnumerable<TransactionData> userTransactionsData = null;
+
+            await _executor.ExecuteWithRetryAsync((System.Func<Task>)(async () =>
+            {
+                userTransactionsData = await _queryApiService.GetAsync<IEnumerable<TransactionData>>( (System.Uri)apiUrlGetAllFriends);
```
Comment (Vovanella95) [inline MoneySplitter.Services/API/TransactionsApiService.cs]: Why you casting to (System.Uri)? Is this property Uri already?

File: MoneySplitter.Services/API/TransactionsApiService.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            }));
+
+            if (userTransactionsData == null)
+            {
+                return result;
+            }
+
+            result.Result = userTransactionsData.Select(x => _mapper.ConvertTransactioDataToTransactionModel(x)).ToList();
```
Comment (Vovanella95) [inline MoneySplitter.Services/API/TransactionsApiService.cs]: I'm not shure, but you can try refactor like this:
result.Result = userTransactionsData.Select(_mapper.ConvertTransactioDataToTransactionModel).ToList();

File: MoneySplitter.Services/Api/QueryApiService.cs [hunk 5]
```diff
@@ -69,6 +69,8 @@ public class QueryApiService : IQueryApiService
             var contentResponce = await responseMessage.Content.ReadAsStringAsync();
             return JsonConvert.DeserializeObject<TContentResponce>(contentResponce);
         }
+
```
Comment (Vovanella95) [inline MoneySplitter.Services/Api/QueryApiService.cs]: Empty lines

File: MoneySplitter.Services/Mapper.cs [hunk 6]
```diff
@@ -43,8 +41,40 @@ public RegisterUserData ConvertRegisterModelToDataRegisterUser( RegisterModel re
...
+        }
 
-            return dataRegisterUser;
+        public RegisterTransactionData ConvertRegisterTransactioModelToRegisterTransactionData(RegisterTransactionModel registerTransactionModel)
+        {
+            return new RegisterTransactionData()
+            {
+                Title = registerTransactionModel.Title,
+                Token=registerTransactionModel.Token,
+                Email = registerTransactionModel.Email,
```
Comment (Vovanella95) [inline MoneySplitter.Services/Mapper.cs]: Please use spaces between =
Title=title - bad
Title = title - good
It is not really important, but looks dirty :)

File: MoneySplitter.Win10/CustomControls/TransctionsControl.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        {
+            get { return (TransactionModel)GetValue(ViewModelProperty); }
+            set { SetValue(ViewModelProperty, value); }
+        }
+        public static readonly DependencyProperty ViewModelProperty = DependencyProperty.Register(
+            "ViewModel",
+            typeof(TransactionModel),
+            typeof(TransctionsControl),
+            null);
+        public TransctionsControl()
```
Comment (Vovanella95) [inline MoneySplitter.Win10/CustomControls/TransctionsControl.xaml.cs]: Insert empty line between methods and DP

File: MoneySplitter.Win10/CustomControls/TransctionsControl.xaml.cs [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+namespace MoneySplitter.Win10.CustomControls
+{
+    public sealed partial class TransctionsControl : UserControl
+    {
+        public TransactionModel ViewModel
+        {
+            get { return (TransactionModel)GetValue(ViewModelProperty); }
+            set { SetValue(ViewModelProperty, value); }
+        }
+        public static readonly DependencyProperty ViewModelProperty = DependencyProperty.Register(
```
Comment (Vovanella95) [inline MoneySplitter.Win10/CustomControls/TransctionsControl.xaml.cs]: Insert empty line here to separate different parts of logic

File: MoneySplitter.Win10/ViewModels/TransactionsViewModel.cs [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+namespace MoneySplitter.Win10.ViewModels
+{
+    public class TransactionsViewModel : Screen
+    {
+        private ObservableCollection<TransactionModel> _transactions;
+
+        private ITransactionsManager _transactionsManager;
+
+        private bool _isNotTransactionsTextVisibility = false;
```
Comment (Vovanella95) [inline MoneySplitter.Win10/ViewModels/TransactionsViewModel.cs]: These properties are false by default

File: MoneySplitter.Win10/ViewModels/TransactionsViewModel.cs [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+        {
+            base.OnActivate();
+
+            IsActiveLoadingProgressRing = true;
+            IsErrorVisible = false;
+            IsNotTransactionsTextVisibility = false;
+
+            var isSuccessExecution = await _transactionsManager.LoadUserTransactions();
+
+            IsActiveLoadingProgressRing = false;
```
Comment (Vovanella95) [inline MoneySplitter.Win10/ViewModels/TransactionsViewModel.cs]: Better rename just to IsLoading

File: MoneySplitter.Services/API/QueryApiService.cs (hunk 1 above)
Comment (Platton96) [reply MoneySplitter.Services/API/QueryApiService.cs]: remove

File: MoneySplitter.Services/API/TransactionsApiService.cs (hunk 3 above)
Comment (Platton96) [reply MoneySplitter.Services/API/TransactionsApiService.cs]: fixed

File: MoneySplitter.Services/API/TransactionsApiService.cs (hunk 2 above)
Comment (Platton96) [reply MoneySplitter.Services/API/TransactionsApiService.cs]: rename

File: MoneySplitter.Services/API/TransactionsApiService.cs (hunk 4 above)
Comment (Platton96) [reply MoneySplitter.Services/API/TransactionsApiService.cs]: refactoring

File: MoneySplitter.Services/Api/QueryApiService.cs (hunk 5 above)
Comment (Platton96) [reply MoneySplitter.Services/Api/QueryApiService.cs]: remove

File: MoneySplitter.Services/Mapper.cs (hunk 6 above)
Comment (Platton96) [reply MoneySplitter.Services/Mapper.cs]: add spaces

File: MoneySplitter.Win10/CustomControls/TransctionsControl.xaml.cs (hunk 7 above)
Comment (Platton96) [reply MoneySplitter.Win10/CustomControls/TransctionsControl.xaml.cs]: add space

File: MoneySplitter.Win10/CustomControls/TransctionsControl.xaml.cs (hunk 8 above)
Comment (Platton96) [reply MoneySplitter.Win10/CustomControls/TransctionsControl.xaml.cs]: add empty lines

**Pull request merged**